approval_polls/static/approval_polls/create.js

```diff
@@ -50,16 +50,19 @@ $(function () {
     }
 
     $('button[id^="confirm-link-"]').click(function() {
-      var buttonId, linkUrl, validUrl;
+      var buttonId, linkUrl, validUrl, urlPattern;
       buttonId = $(this).attr('id');
       buttonId = buttonId.split('-').pop();
       linkUrl = $('#url-' + buttonId).val();
       linkUrl = $.trim(linkUrl);
-      // Copyright (c) 2010-2013 Diego Perini, MIT licensed
-      // https://gist.github.com/dperini/729294
-      // see also https://mathiasbynens.be/demo/url-regex
-      // modified to allow protocol-relative URLs
-      validUrl = /^(?:(?:(?:https?|ftp):)?\/\/)(?:\S+(?::\S*)?@)?(?:(?!(?:10|127)(?:\.\d{1,3}){3})(?!(?:169\.254|192\.168)(?:\.\d{1,3}){2})(?!172\.(?:1[6-9]|2\d|3[0-1])(?:\.\d{1,3}){2})(?:[1-9]\d?|1\d\d|2[01]\d|22[0-3])(?:\.(?:1?\d{1,2}|2[0-4]\d|25[0-5])){2}(?:\.(?:[1-9]\d?|1\d\d|2[0-4]\d|25[0-4]))|(?:(?:[a-z\u00a1-\uffff0-9]-*)*[a-z\u00a1-\uffff0-9]+)(?:\.(?:[a-z\u00a1-\uffff0-9]-*)*[a-z\u00a1-\uffff0-9]+)*(?:\.(?:[a-z\u00a1-\uffff]{2,})).?)(?::\d{2,5})?(?:[/?#]\S*)?$/i.test(linkUrl);
+      // http://stackoverflow.com/questions/5717093/check-if-a-javascript-string-is-an-url
+      urlPattern = new RegExp('^(https?:\\/\\/)?'+ // protocol
+      '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|'+ // domain name
+      '((\\d{1,3}\\.){3}\\d{1,3}))'+ // OR ip (v4) address
+      '(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*'+ // port and path
+      '(\\?[;&a-z\d%_.~+=-]*)?'+ // query string
+      '(\\#[-a-z\\d_]*)?$','i'); // fragment locater
```

Comment:
@kushankr, would it be possible to relax this condition a bit so that it would accept different types of URLs including google searches. I was wondering if it should accept any link that is valid from the address bar. 
I was testing out the deployment when i realized it was not linking to the following URLs: 
http://www.parks.ca.gov/?page_id=468
https://en.wikipedia.org/wiki/Angel_Island_(California)
https://www.google.com/webhp?sourceid=chrome-instant&ion=1&espv=2&ie=UTF-8#q=alcatraz

Please let me know what you think.